### PR TITLE
feat: added MNIST, Fashion-MNIST and KMNIST datasets

### LINKED ISSUE #161: Add the MNIST Dataset
### Is your feature request related to a problem?

The often used MNIST dataset would be nice to have.

### Desired solution

Add the [MNIST](http://yann.lecun.com/exdb/mnist/) dataset.

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### LINKED ISSUE #162: Add the Fashion-MNIST Dataset
### Is your feature request related to a problem?

Because the MNIST dataset is considered too simple as CNNs can achive almost 100% accuracy it would be nice to have another MNIST like dataset that is more difficult.

### Desired solution

Add the [Fashion-MNIST](https://github.com/zalandoresearch/fashion-mnist) dataset.

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### LINKED ISSUE #163: Add the KMNIST Dataset
### Is your feature request related to a problem?

It would be nice to have another different MNIST like Dataset. KMNIST is a dataset containing of Japanese letters.

### Desired solution

Add the [Kuzushiji-MNIST](https://github.com/rois-codh/kmnist) dataset.

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by Marsmaennchen221**
Closes #161 
Closes #162 
Closes #163 

### Summary of Changes

feat: added MNIST, Fashion-MNIST and KMNIST datasets
build: bump safe-ds to ^0.24.0

### CODE REVIEW

**Comment by Marsmaennchen221** [comment]
@lars-reimann Currently, the tests download each dataset once. Therefore, the tests take a while. Do you have an idea what we could do about that?

**Comment by lars-reimann** [comment]
> @lars-reimann Currently, the tests download each dataset once. Therefore, the tests take a while. Do you have an idea what we could do about that?

https://github.com/actions/cache could help here.

File: src/safeds_datasets/image/_mnist/_mnist.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+def load_kmnist(path: str | Path, download: bool = True) -> tuple[ImageDataset[Column], ImageDataset[Column]]:
+    """
+    Load the `Kuzushiji-MNIST <https://github.com/rois-codh/kmnist>`_ datasets.
+
+    Parameters
+    ----------
+    path:
+        the path were the files are stored or will be downloaded to
+    download:
+        weather the files should be downloaded to the given path
```
Comment (lars-reimann) [inline src/safeds_datasets/image/_mnist/_mnist.py:152]:
```suggestion
        whether the files should be downloaded to the given path
```

File: src/safeds_datasets/image/_mnist/_mnist.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+def load_fashion_mnist(path: str | Path, download: bool = True) -> tuple[ImageDataset[Column], ImageDataset[Column]]:
+    """
+    Load the `Fashion-MNIST <https://github.com/zalandoresearch/fashion-mnist>`_ datasets.
+
+    Parameters
+    ----------
+    path:
+        the path were the files are stored or will be downloaded to
+    download:
+        weather the files should be downloaded to the given path
```
Comment (lars-reimann) [inline src/safeds_datasets/image/_mnist/_mnist.py:112]:
```suggestion
        whether the files should be downloaded to the given path
```

File: src/safeds_datasets/image/_mnist/_mnist.py [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+def load_mnist(path: str | Path, download: bool = True) -> tuple[ImageDataset[Column], ImageDataset[Column]]:
+    """
+    Load the `MNIST <http://yann.lecun.com/exdb/mnist/>`_ datasets.
+
+    Parameters
+    ----------
+    path:
+        the path were the files are stored or will be downloaded to
+    download:
+        weather the files should be downloaded to the given path
```
Comment (lars-reimann) [inline src/safeds_datasets/image/_mnist/_mnist.py:72]:
```suggestion
        whether the files should be downloaded to the given path
```

**Comment by Marsmaennchen221** [comment]
> > @lars-reimann Currently, the tests download each dataset once. Therefore, the tests take a while. Do you have an idea what we could do about that?
> 
> https://github.com/actions/cache could help here, if the data is not too large. We have 10 GB for caches in total.

Ok I will look at this at a later time. For now, this pr is ready.

**Comment by lars-reimann** [comment]
OK, thanks. Review will take a while. I have other priorities right now.

**Review by lars-reimann — APPROVED** [review]

**Pull request merged**